### ISSUE TITLE
Problems when searching for text

### ISSUE THREAD

**Issue by gjtorikian** [ajaxorg/ace#683]
This one is kind of hard to explain; I tried to summarize what I found in this comment: https://github.com/ajaxorg/cloud9/commit/91311fedd1d97860124be667058a4467e18ad016#L333

Okay, so let's say the doc has the string `juhu`. If you search for `j`, the `j` in `juhu` is highlighted. Great.

However, let's say you now decide to search for `ju`. Since `j` is highlighted, you would think that it would match the `ju` in `juhu`, but that's not what happens. The search fails. Actually, if you chose instead to find `u`, the `u` would then properly highlight. If, whilst on `j`, you chose to search for `uh`, that too would highlight.

Basically, even though the selection is on `j`, the _cursor_ is after `j`, so subsequent string searches provide awkward behaviors. This actually may be very well intended, so perhaps there should be a new property/option for search that overrides this.

The workaround in C9 is to check if the highlighted range (`j`) is the start of the next search term (`ju`); if it is, the cursor shifts back one char, so that the search for `ju` properly highlights `ju`. This is really used for a "quicksearch"-like implementation, as we've used it.
